cmd/derper/derper.go

```diff
@@ -235,6 +235,14 @@ func main() {
 			cert.Certificate = append(cert.Certificate, s.MetaCert())
 			return cert, nil
 		}
+		httpsrv.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// Security scanners get cranky when HTTPS sites don't set
+			// HSTS. Set it even though derper doesn't really serve
+			// anything of interest to browsers (and API clients like
+			// tailscale don't obey HSTS).
+			w.Header().Set("Strict-Transport-Security", "max-age=600; includeSubDomains")
```

Comment:
only do this is r.TLS != nil ?

Line 249 won't use this, though, right?

Reply:
(Do the HSTS scanners expect to see it on HTTP responses too?)

If so, you'll need to teach this code for which domain names to include it.

Reply:
This is already in an `if serveTLS` block.

And yes, it wraps the TLS handler only, so port 80 continues to serve without HSTS directives (and besides, HSTS says browsers must ignore HSTS directives sent over HTTP anyway, because it's a DoS vector).

Reply:
LGTM

Reply:
To clarify: no, HSTS headers MUST NOT be set on HTTP responses. And if HSTS headers are present in HTTP responses, browsers MUST ignore them, because otherwise a mitm could force-enable HSTS on your site by editing your HTTP traffic, which would lead to a DoS if your site doesn't do HTTPS.

So, I think this change is correct: when answering HTTPS requests, set HSTS headers; otherwise, don't.